src/test/java/io/confluent/examples/streams/WordCountLambdaIntegrationTest.java

```diff
@@ -59,78 +56,69 @@
+          outputTopic, topologyTestDriver, new StringDeserializer(), new LongDeserializer());
+      assertThat(actualWordCounts).isEqualTo(expectedWordCounts);
+    }
+  }
 
+  private StreamsBuilder createTopology() {
+    final StreamsBuilder builder = new StreamsBuilder();
     final KStream<String, String> textLines = builder.stream(inputTopic);
-
     final Pattern pattern = Pattern.compile("\\W+", Pattern.UNICODE_CHARACTER_CLASS);
-
     final KTable<String, Long> wordCounts = textLines
-      .flatMapValues(value -> Arrays.asList(pattern.split(value.toLowerCase())))
-      // no need to specify explicit serdes because the resulting key and value types match our default serde settings
-      .groupBy((key, word) -> word)
-      .count();
-
-    wordCounts.toStream().to(outputTopic, Produced.with(stringSerde, longSerde));
+        .flatMapValues(value -> Arrays.asList(pattern.split(value.toLowerCase())))
+        // no need to specify explicit serdes because the resulting key and value types match our default serde settings
+        .groupBy((key, word) -> word)
+        .count();
+    wordCounts.toStream().to(outputTopic, Produced.with(Serdes.String(), Serdes.Long()));
+    return builder;
+  }
 
-    try (final TopologyTestDriver topologyTestDriver = new TopologyTestDriver(builder.build(), streamsConfiguration)) {
-      //
-      // Step 2: Produce some input data to the input topic.
-      //
-      IntegrationTestUtils.produceKeyValuesSynchronously(
-        inputTopic,
-        inputValues.stream().map(v -> new KeyValue<>(null, v)).collect(Collectors.toList()),
-        topologyTestDriver,
-        new IntegrationTestUtils.NothingSerde<>(),
-        new StringSerializer()
-      );
+  private Properties createTopologyConfiguration() {
 
-      //
-      // Step 3: Verify the application's output data.
-      //
-      final Map<String, Long> actualWordCounts = IntegrationTestUtils.drainTableOutput(
-        outputTopic,
-        topologyTestDriver,
-        new StringDeserializer(),
-        new LongDeserializer()
-      );
-      assertThat(actualWordCounts).isEqualTo(expectedWordCounts);
-    }
+    final Properties streamsConfiguration = new Properties();
+    streamsConfiguration.put(StreamsConfig.APPLICATION_ID_CONFIG, "wordcount-lambda-integration-test");
+    streamsConfiguration.put(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, "dummy config");
+    streamsConfiguration.put(StreamsConfig.DEFAULT_KEY_SERDE_CLASS_CONFIG, Serdes.String().getClass().getName());
+    streamsConfiguration.put(StreamsConfig.DEFAULT_VALUE_SERDE_CLASS_CONFIG, Serdes.String().getClass().getName());
+    // The commit interval for flushing records to state stores and downstream must be lower than
+    // this integration test's timeout (30 secs) to ensure we observe the expected processing results.
+    streamsConfiguration.put(StreamsConfig.COMMIT_INTERVAL_MS_CONFIG, TimeUnit.SECONDS.toMillis(10));
```

Comment:
this config should not be necessary with TTD.

Reply:
Ack